File: uniapp_flutter/lib/utils/media_subtype.dart
enum MediaSubType {
  FULL,
  PREVIEW,
  THUMBNAIL
}

class MediaSubTypeHelper {
  static int getValue(MediaSubType status){
    switch(status){
      case MediaSubType.FULL:
        return 1;
      case MediaSubType.PREVIEW:
        return 2;
      case MediaSubType.THUMBNAIL:
        return 3;
      default:
        return 0;
    }
  }
  static final Map<int, MediaSubType> valueToStateNameMap = new Map();
  final int value = 0;
  static createMap() {

    for (MediaSubType myEnum in MediaSubType.values) {
      valueToStateNameMap[MediaSubTypeHelper.getValue(myEnum)]= myEnum;
    }
  }

  static MediaSubType getStateByValue(int value) {
    if(valueToStateNameMap.isEmpty) {
      createMap();
    }
    return valueToStateNameMap[value];
  }
}

File: app/app/src/main/java/com/vassar/unifiedapp/utils/MediaSubType.java
package com.vassar.unifiedapp.utils;
import java.util.HashMap;
import java.util.Map;

public enum MediaSubType {

    FULL(0),
    PREVIEW(1),
    THUMBNAIL(2);


    private static final Map<Integer, MediaSubType> valueToSubTypeNameMap = new HashMap<>();
    private final Integer value;

    static {
        for (MediaSubType myEnum : values()) {
            valueToSubTypeNameMap.put(myEnum.getValue(), myEnum);
        }
    }

    private MediaSubType(final Integer newValue) {
        value = newValue;
    }

    public Integer getValue() {
        return value;
    }

    public static MediaSubType getSubTypeByValue(Integer value) {
        return valueToSubTypeNameMap.get(value);
    }
}
